src/client.rs

```diff
@@ -58,15 +53,15 @@ impl<T: Request> IntoFuture for Route<T> {
     }
 }
 
-impl<T> Deref for Route<T> {
-    type Target = T;
+impl Deref for Route<GenerateContent> {
+    type Target = GenerateContent;
```

Comment:
This is also perhaps a little questionable. Removing the generic lets me add a specific `Deref`/`DerefMut` impl for `StreamGenerateContent`, and it didn't cause any problems compiling, but I might be assuming something about the desired API for the library 🤷